### PR TITLE
feat: search experiment feed header

### PR DESCRIPTION
**Pull request by sshanzel**
## Changes

Previews:
- There are a lot of changes on this PR though it was only revolving around the feed header change.
- Due to the large differences between the header for control and v1 values for the experiment, we have decided to introduce separate components to render based on version rather than having numerous conditions on the classes based on the said experiment.
- The changes related to `isSearchOn` is just reverting some of the changes in the intermediate branch, but they are basically the legacy code.
- Moved the `FeedFilters` modal to be under the `LazyModal` usage.
- The dedicated search page would still render based on the flag. Both variants will be under the same URL.

Screenshot 2023-08-15 at 6 16 39 PM
Screenshot 2023-08-15 at 5 46 42 PM
Screenshot 2023-08-15 at 5 46 32 PM
Screenshot 2023-08-15 at 5 46 13 PM

### Describe what this PR does
- Short and concise, bullet points can help
- Screenshots if applicable can also help

## Events

Did you introduce any new tracking events?
Don't forget to update the [Analytics Taxonomy sheet](https://docs.google.com/spreadsheets/d/18Lv7zXges9QfVX5VYL1a-Hyl0e1sQ3sLr0OK8YZWKXI/edit#gid=0)

Log the new/changed events below:

| Type   | event_name  | value |
|--------|-------------|-------|
| Change/New | event name  | extra: { ... } |

### On those affected packages:
- [ ] Have you done sanity checks in the webapp?
- [ ] Have you done sanity checks in the extension?
- [ ] Does this not break anything in companion?

### Did you test the modified components media queries?
- [ ] MobileL (420px)
- [ ] Tablet (656px)
- [ ] Laptop (1020px)

#### Did you test on actual mobile devices?
- [ ] iOS (Chrome and Safari)
- [ ] Android

WT-1639 #done

### CODE REVIEW

File: packages/shared/src/components/feeds/FeedContainer.tsx [hunk 1]
```diff
@@ -122,10 +139,33 @@ export const FeedContainer = ({
...
+                  field: 'w-full',
+                }}
+                showProgress={false}
+                onSubmit={onSearch}
+              />
+              {besideSearch}
+            </FlexRow>
+          )}
+          {isV1Search && (
+            <FlexRow className="flex flex-1 mt-4">
```
Comment (rebelchris) [inline packages/shared/src/components/feeds/FeedContainer.tsx:156]: Still not a fan of these 😂
Why don't we simply pass the row, these are super prone to changes down the line and just don't really make sense at this point.

File: packages/shared/src/components/filters/MyFeedHeading.tsx [hunk 2]
```diff
@@ -34,21 +42,42 @@ function MyFeedHeading({
...
+      wrapper: 'mr-auto z-modal',
+    },
+    message: filterAlertMessage,
+    placement:
+      sidebarRendered && searchVersion === SearchExperiment.Control
+        ? AlertPlacement.Right
+        : AlertPlacement.Bottom,
+  };
+
+  if (searchVersion === SearchExperiment.V1) {
```
Comment (rebelchris) [inline packages/shared/src/components/filters/MyFeedHeading.tsx:64]: We do render for both right?

File: packages/shared/src/components/sidebar/DiscoverSection.tsx [hunk 3]
```diff
@@ -57,16 +61,19 @@ export function DiscoverSection({
         </span>
       ),
     },
-    {
+  ];
+
+  if (searchValue !== SearchExperiment.V1) {
```
Comment (rebelchris) [inline packages/shared/src/components/sidebar/DiscoverSection.tsx:66]: Should we not say if it's explicitly control?
What happens if we introduce V2 for instance

File: packages/shared/src/hooks/feed/useFeedName.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+export const useFeedName = ({
+  feedName,
+  isSearchOn,
+}: UseFeedNameProps): UseFeedName => {
+  const searchVersion = useFeature(Features.Search);
+
+  return {
+    isUpvoted:
+      searchVersion === SearchExperiment.Control
+        ? !isSearchOn && feedName === 'upvoted'
```
Comment (rebelchris) [inline packages/shared/src/hooks/feed/useFeedName.ts:27]: Can't we rather abstract the logic above?
To say isSearchOn is false//null when it's not control?

**Review by rebelchris — CHANGES_REQUESTED** [review]
Loads of logic 🤯
Have some generic feedback, can we somehow maybe abstract logic so we don't have so much conditional stuff?

Else I'm already praying for a removal task 😂

File: packages/shared/src/components/filters/MyFeedHeading.tsx (hunk 2 above)
Comment (sshanzel) [reply packages/shared/src/components/filters/MyFeedHeading.tsx:64]: Yes, we have the default that starts at line 77.

File: packages/shared/src/components/sidebar/DiscoverSection.tsx (hunk 3 above)
Comment (sshanzel) [reply packages/shared/src/components/sidebar/DiscoverSection.tsx:66]: Makes sense. Let's do that.

File: packages/shared/src/hooks/feed/useFeedName.ts (hunk 4 above)
Comment (sshanzel) [reply packages/shared/src/hooks/feed/useFeedName.ts:27]: That `isSearchOn` is coming from way up the tree based on a state. Can you provide a simple example of how you envision it to be?

File: packages/shared/src/components/feeds/FeedContainer.tsx (hunk 1 above)
Comment (rebelchris) [reply packages/shared/src/components/feeds/FeedContainer.tsx]: Yes correct :) Think we should just remove them to be honest.

Comment (sshanzel) [reply packages/shared/src/components/feeds/FeedContainer.tsx]: Amazing. Should be removed by now.

**Comment by sshanzel** [comment]
@rebelchris I've now added the Cloudinary links. Should be ready for final review 🚢

File: packages/shared/src/lib/image.ts [hunk 5]
```diff
@@ -1,4 +1,14 @@
 export const cloudinary = {
+  feed: {
+    bg: {
+      laptop:
+        'https://daily-now-res.cloudinary.com/image/upload/v1665224467/public/Desktop-ask-bg.png',
```
Comment (rebelchris) [inline packages/shared/src/lib/image.ts:5]: All of these seem to be wrong, missing the f_auto can you ensure with design please?
Or I can help get them if not

**Review by rebelchris — CHANGES_REQUESTED** [review]
One small request, otherwise looks great to me

File: packages/shared/src/lib/image.ts (hunk 5 above)
Comment (sshanzel) [reply packages/shared/src/lib/image.ts:5]: They shared one with it, but it wasn't working, not sure if the files are not yet public. Should be found here: https://dailydotdev.slack.com/archives/C05EM7ZFQMQ/p1692271488033119?thread_ts=1692265064.294219&cid=C05EM7ZFQMQ

**Comment by sshanzel** [comment]
Merged so we can now test on preview. Will create a ticket for updating cloudinary links.

**Pull request merged**